Sources/DVR/SessionDataTask.swift

```diff
@@ -42,30 +42,35 @@ final class SessionDataTask: URLSessionDataTask {
     }
 
     override func resume() {
-        let cassette = session.cassette
-
-        // Find interaction
-        if let interaction = session.cassette?.interactionForRequest(request, headersToCheck: headersToCheck) {
-            self.interaction = interaction
-            // Forward completion
-            if let completion = completion {
-                queue.async {
-                    completion(interaction.responseData, interaction.response, nil)
+        
+        if session.recordMode == .none {
+            let cassette = session.cassette
+
+            // Find interaction
+            if let interaction = session.cassette?.interactionForRequest(request, headersToCheck: headersToCheck) {
+                self.interaction = interaction
+                // Forward completion
+                if let completion = completion {
+                    queue.async {
+                        completion(interaction.responseData, interaction.response, nil)
+                    }
                 }
+                session.finishTask(self, interaction: interaction, playback: true)
+                return
             }
-            session.finishTask(self, interaction: interaction, playback: true)
-            return
-        }
 
-        if cassette != nil {
-            fatalError("[DVR] Invalid request. The request was not found in the cassette.")
-        }
+            if cassette != nil {
+                fatalError("[DVR] Invalid request. The request was not found in the cassette.")
+            }
 
-        // Cassette is missing. Record.
-        if session.recordingEnabled == false {
-            fatalError("[DVR] Recording is disabled.")
+            // Cassette is missing. Record.
+            if session.recordingEnabled == false {
+                fatalError("[DVR] Recording is disabled.")
+            }
```

Comment:
I find this pretty hard to reason about. I would suggest:

```swift
// interaction not found
if cassette !=nil {
   fatalError(... request not in cassette)
} else {
  // if request not found and recording is not enabled, error
  if session.recordingEnabled == false {
    fatalError(...)
  }
}
```

Of course, this assumes recordingEnabled is set property based on the recordMode.